shell/config/router/navigation-guards/history.js

```diff
@@ -0,0 +1,13 @@
+export function install(router, context) {
+  router.beforeEach((to, from, next) => loadHistory(to, from, next, context));
+}
+
+export async function loadHistory(to, from, next) {
+  // GC should be notified of route change before any find/get request is made that might be used for that page
```

Comment:
Sorry for the question, but what is GC? 
Beside understanding we have other issues in the find/get logic, which should need an issue, why do we set `window._popStateDetected`? If also this is an existing hack, could we please flag it with a TODO and brief explanation?
I am sorry to ask you this, but we are already lacking knowledge of most of the code, so if we could track it that would be an improvement.
Once again, if you plan to work on this in a short term, forget it 😅

Reply:
> Sorry for the question, but what is GC?
Performance related:
![image](https://github.com/rancher/dashboard/assets/55104481/680ada2d-1ac1-46f6-82c2-2fbfcec254fe)

This comment looks like it was decoupled from the code it should represent. I moved it back to where it should be.

> Beside understanding we have other issues in the find/get logic, which should need an issue, why do we set window._popStateDetected? 

I've added a todo and issue:
https://github.com/rancher/dashboard/issues/11274

Reply:
Thank you for the clarification!